frontend/src/components/module-editor/LessonItem.tsx

```diff
@@ -16,6 +21,44 @@ const LessonItem = ({
 }: OptionsProps): React.ReactElement => {
   const [isHovered, setIsHovered] = useState(false);
 
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [title, setTitle] = useState(text);
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const { courseID, moduleIndex }: ModuleEditorParams = useParams();
+  const moduleID = parseInt(moduleIndex, 10);
+
+  const context = useContext(EditorContext);
+  if (!context) return <></>;
+  const { state, dispatch } = context;
+
+  const { course } = state;
+
+  const resetState = () => {
+    setTitle("");
+    setErrorMessage("");
+    setIsInvalid(false);
+  };
+
+  const updateLessonName = (lessonTitle: string) => {
+    if (title) {
+      dispatch({
+        type: "update-lesson",
+        value: {
+          course: courseID,
+          module: course.modules[moduleID].id,
+          title: lessonTitle,
+          content: [],
```

Comment:
Unfortunately, dispatching the action like this clears the lesson content, which is something we definitely don't want. Could you create a new action type instead which only updates the title?